test/test_kubeclient.rb

```diff
@@ -341,6 +385,74 @@ def test_get_all
     assert_instance_of(Kubeclient::ServiceAccount, result['service_account'][0])
   end
 
+  def test_get_all_raw
+    stub_request(:get, %r{/api/v1$})
+      .to_return(body: open_test_file('core_api_resource_list.json'), status: 200)
+
+    stub_request(:get, %r{/bindings})
+      .to_return(body: open_test_file('bindings_list.json'), status: 404)
```

Comment:
why 404?

Reply:
¯\\(°_o)/¯

I basically did a copy pasta from the test above, and the stubs are what was done there.  The 404 was something I questioned myself, but was unsure why it was there.  Just left it.

Reply:
and why didn't it fail the test?  bindings is not in tested array below, so either drop this stub or add it?

Reply:
This is where I disagree, but to do so, I have to take some guesses into what the original `test_get_all` test above this one was attempting to accomplish originally (again, I didn't write this, just copied, so I am definitely taking a guess here).

To start, `test_get_all` also doesn't test against the `bindings` result, and I think that is in part to why the `404` was added in the first place.  There is a portion of that method that actually skips over any errors in that might have occurred when fetching everything:

https://github.com/abonas/kubeclient/blob/30b8285/lib/kubeclient/common.rb#L362

That means that a `404` (or I assume even a `500`), could occur, but it would allow the rest of the requests to be fetched and the results that were successful returned normally.  Whether or not this is the _**#right™**_ implementation is another story, but discussing that is probably out of scope for this PR.

I wanted to make sure that when testing any of these methods where the only change was the addition of the `as: :raw`, that the functionality mirrored what was being tested previously, and the only change was the output format.  Hence why most of these tests are carbon copies of tests that came prior to them, but have different assertions because the output differs.

Regarding making this DRY, as elaborated in my [other answer](https://github.com/abonas/kubeclient/pull/262#discussion_r135651199), I am fine doing this if that is what you would prefer, but didn't do so to start.  Waiting on a green light though before doing that.

Reply:
agree to not refactor here, LGTM.
thanks for the detailed explanation, I didn't realize `all_entities` ignores errors, it makes sense now.
sorry for side-tracking this.

Reply:
> sorry for side-tracking this.

No worries.  As I said, I had the same thought as you regarding the refactoring initially, but came to this conclusion for the time being.  I will probably open a PR just for some test refactoring regarding some of the duplicate code I generated after this gets merged 😄